dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/BaseAnalyticalObject.java

```diff
@@ -1424,8 +1426,14 @@ public Set<String> getSubscribers()
         return subscribers;
     }
 
+    @CheckForNull
     public void setSubscribers( Set<String> subscribers )
     {
+        if ( subscribers == null )
+        {
+            subscribers = new HashSet<>();
+        }
+
```

Comment:
I think we should not change the input parameter, rather check if it's null then set this.subscribers to new HashSet.

Reply:
Good point, it will be safer.